hail/src/main/scala/is/hail/io/CodecSpec.scala

```diff
@@ -16,6 +16,7 @@ import org.apache.spark.rdd.RDD
 trait AbstractTypedCodecSpec extends Spec {
   def encodedType: EType
   def encodedVirtualType: Type
+  def encodedPhysicalType: PType
```

Comment:
I think this is where we want to go? Can't pass along requiredeness of data to partitions, unless carried along on EType or PType, and TypedCodecSpec's most oft-used constructor takes a PType, so threading PTypes through is a minimal change.